src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt

```diff
@@ -76,6 +78,59 @@ class WLSService(
         return movedItem
     }
 
+    override suspend fun pickItems(
+        hostName: HostName,
+        itemsPickedMap: Map<String, Double>
+    ) {
+        val itemIds =
+            itemsPickedMap.map {
+                ItemId(hostName, it.key)
+            }
+
+        if (!itemRepository.doesEveryItemExist(itemIds)) {
+            throw ItemNotFoundException("Some items do not exist in the database, and were unable to be picked")
+        }
+
+        itemIds.map { itemId ->
+            val item = getItem(itemId.hostName, itemId.hostId)!!
+            if (itemsPickedMap[item.hostId] == null) {
+                // TODO - Review this. This state should generally not happen with the preconditions. Does it need to be handled at all?
+                throw ItemNotFoundException("Item with ID ${itemId.hostId} for host ${item.hostName} was not found, despite existing!")
+            }
+            val movedItem =
+                itemRepository.moveItem(
+                    item.hostId,
+                    item.hostName,
+                    item.quantity?.minus((itemsPickedMap[item.hostId] ?: 0.0)) ?: 0.0,
+                    "Picked"
```

Comment:
Location should be WITH_LENDER or similar and only if quantity is 0.0, also shouldn't you check if quantity is negative after subtraction?

Reply:
Should we declare some statics for locations somewhere?
Also yes, we do need a check here. I forgot that this isn't being passed to WLSServices `moveItem` method. 
What should we do in the scenario where someone has picked more items than we have registered? Be it invalid state, or another mismatch in inventory

Reply:
I am handling this by logging the error and clamping the quantity. We should still consider implementing further warnings in case this happens (send some notification somewhere?)